src/main/kotlin/inc/evil/d17map/security/authentication/AuthService.kt

```diff
@@ -0,0 +1,70 @@
+package inc.evil.d17map.security.authentication
+
+import inc.evil.d17map.entities.User
+import inc.evil.d17map.exceptions.RoleNotFoundException
+import inc.evil.d17map.exceptions.UserAlreadyExistsException
+import inc.evil.d17map.mappers.toRoleResponse
+import inc.evil.d17map.repositories.UserRepository
+import inc.evil.d17map.security.authentication.jwt.TokenProvider
+import inc.evil.d17map.security.authorization.RoleRepository
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.AuthenticationException
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Service
+
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class AuthService(
+    private val userRepository: UserRepository,
+    private val roleRepository: RoleRepository,
+    private val passwordEncoder: PasswordEncoder,
+    private val authenticationManager: AuthenticationManager,
+    private val tokenProvider: TokenProvider,
+
+    ) {
+    companion object {
+        const val DEFAULT_ROLE = "ROLE_STUDENT"
+    }
+
+    fun registerUser(registerRequest: AuthRequest) {
+        logger.info { "Checking if user $registerRequest already exists." }
```

Comment:
Może nie loguj całego obiektu authRequest (zawiera hasło), tylko np samo username?

Reply:
O kurczaczki, przegapiłem to 🫨